assets/dev/js/editor/components/template-library/manager.js

```diff
@@ -471,6 +471,11 @@ const TemplateLibraryManager = function() {
 		self.setFilter( 'type', args.type, true );
 		self.setFilter( 'subtype', args.subtype, true );
 
+		if ( 'cloud' === args.source && ! elementor.config.library_connect.is_connected ) {
+			self.layout.showCloudConnectView();
+			return;
+		}
```

Comment:
Interjecting here already so we can avoid an unnecessary AJAX call to `?source=cloud` when the modal originally opens or the user clicks on the tab.